components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -49,6 +60,35 @@ class AddonManager(
             throw AddonManagerException(throwable)
         }
     }
+
+    /**
+     * Try updates the provided [extensionId] by updating it in the the [store] and re-attaching
+     * all the handlers.
+     */
+    fun tryUpdateExtension(extensionId: String, onFinish: ((Status) -> Unit)) {
+        val extension = WebExtensionSupport.installedExtensions[extensionId]
+
+        if (extension == null) {
+            onFinish(Status.NotInstalled)
```

Comment:
I find `Status.NotInstalled` a bit out of the scope for an update function vs. `tryUpdateExtension` taking in `extension` as the parameter, and `WebExtensionSupport.installedExtensions[extensionId]` and null check happening outside. 🤔

Reply:
I'm using `Status.NotInstalled` to double check, we could just ignore when we don't find in `installedExtensions`  
 
When an extension is installed, we send the `extensionId` to the `workmanager`'s jobs database. It periodically calls `AddonUpdaterWorker` that calls `tryUpdateExtension`.

A situation where the user uninstalls the `extension` while a scheduled work was already in progress causing that the `extension` is not in `installedExtensions`. I'm using  `NotInstalled` to let the callers take action about it. When this happens`AddonUpdaterWorker` is  removing the job from the `workmanager`'s jobs database.